components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -4836,6 +5085,156 @@ public void deleteApplication(Application application) throws APIManagementExcep
+     * @param organization Organization
+     * @throws APIManagementException if failed to delete application list
+     */
+    public void deleteApplicationList(String organization) throws APIManagementException {
+
+        Connection connection = null;
+        PreparedStatement deleteMappingQuery = null;
+        PreparedStatement prepStmtGetConsumerKey = null;
+        PreparedStatement deleteRegistrationQuery = null;
+        PreparedStatement deleteSubscription = null;
+        PreparedStatement deleteDomainApp = null;
+        PreparedStatement deleteAppKey = null;
+        PreparedStatement deleteApp = null;
+        ResultSet rs = null;
+
+        String getConsumerKeyQuery = SQLConstants.GET_CONSUMER_KEYS_OF_APPLICATION_LIST_SQL;
+
+        String deleteSubscriptionsQuery = SQLConstants.REMOVE_APPLICATION_LIST_FROM_SUBSCRIPTIONS_SQL;
+
+        String deleteApplicationKeyQuery = SQLConstants.REMOVE_APPLICATION_LIST_FROM_APPLICATION_KEY_MAPPINGS_SQL;
+
+        String deleteDomainAppQuery = SQLConstants.REMOVE_APPLICATION_FROM_DOMAIN_MAPPINGS_SQL;
+
+        String deleteApplicationQuery = SQLConstants.REMOVE_APPLICATION_LIST_FROM_APPLICATIONS_SQL;
+
+        String deleteRegistrationEntry = SQLConstants.REMOVE_APPLICATION_LIST_FROM_APPLICATION_REGISTRATIONS_SQL;
+
+        boolean transactionCompleted = true;
+        try {
+            connection = APIMgtDBUtil.getConnection();
+            connection.setAutoCommit(false);
+
+            if (multiGroupAppSharingEnabled) {
+                transactionCompleted = updateGroupIDMappingsBulk(connection, organization);
+            }
+
+            prepStmtGetConsumerKey = connection.prepareStatement(getConsumerKeyQuery);
+            prepStmtGetConsumerKey.setString(1, organization);
+
+            rs = prepStmtGetConsumerKey.executeQuery();
+
+            deleteDomainApp = connection.prepareStatement(deleteDomainAppQuery);
+
+            while (rs.next()) {
+                String consumerKey = rs.getString(APIConstants.FIELD_CONSUMER_KEY);
+                String keyManagerName = rs.getString("NAME");
+                String keyManagerOrganization = rs.getString("ORGANIZATION");
+
+                // This is true when OAuth App has been created by pasting consumer key/secret in the screen.
+                String mode = rs.getString("CREATE_MODE");
+                if (consumerKey != null) {
+
+                    deleteDomainApp.setString(1, consumerKey);
+                    deleteDomainApp.addBatch();
+                    KeyManager keyManager = KeyManagerHolder.getKeyManagerInstance(keyManagerOrganization,
+                            keyManagerName);
+
+                    if (keyManager != null) {
+                        try {
+                            keyManager.deleteMappedApplication(consumerKey);
+                        } catch (APIManagementException e) {
+                            handleException("Error while Deleting Client Application", e);
```

Comment:
Shouldn't we log and proceed?
We need to keep the failing application details to send as the response too. We can have a DeletionResponse object as the return type and keep filling it until finish.
Please check this in all places.

Make sure to include the organizationId in all error/debug logs to make it easier to identify which organization the failure log belong to when someone analyzing logs.
Please fix this in all places.